src/app/functions/universal/guidedExperienceBreaches.ts

```diff
@@ -6,6 +6,19 @@ import { BreachDataTypes } from "./breach";
 import { SubscriberBreach } from "../../../utils/subscriberBreaches";
 import { GuidedExperienceBreaches } from "../server/getUserBreaches";
 
+function isUnresolvedDataBreachClass(
+  breach: SubscriberBreach,
+  breachDataClass: (typeof BreachDataTypes)[keyof typeof BreachDataTypes],
+): boolean {
+  const affectedDataClasses = breach.dataClassesEffected.map(
+    (dataClass) => Object.keys(dataClass)[0],
```

Comment:
Why are we only looking at the first affected data class here? Shouldn't every class be relevant?

(The data structure of `.dataClassesEffected` is so funky, I get lost whenever I try to parse code that references it...)

Reply:
`.dataClassesEffected` has the type `DataClassEffected[]`. So it can look something like …
```
[
  { "firstDataClassKey": 1 },
  { "secondDataClassKey": 2 },
]
```

… and they are just being converted to an array `["firstDataClassKey",  "secondDataClassKey"]`.

Reply:
Right, but if it holds something like

```ts
[
  { "firstDataClassKey": 1, "thirdDataClassKey": 42 },
  { "secondDataClassKey": 2 },
]
```

Then the `[0]` would result in `thirdDataClassKey` (or potentially `firstDataClassKey`, as I don't think sort order is guaranteed) being dropped.

(This is what I meant by https://github.com/mozilla/blurts-server/pull/3791#discussion_r1404559596.)

Reply:
Ah, I see: No that is not the case. I’m not sure what the origin of deciding to go with this data structure was. @mansaj will probably know better.